packages/core/src/util/ng_dev_mode.ts

```diff
@@ -115,7 +115,7 @@ export function initNgDevMode(): boolean {
     if (typeof ngDevMode !== 'object') {
       ngDevModeResetPerfCounters();
     }
-    return !!ngDevMode;
+    return typeof ngDevMode !== 'undefined' && ngDevMode;
```

Comment:
It looks like the `!!` is needed to make sure the return value type is aligned with function type signature (there are some tests that are failing on CI because of that):
```suggestion
    return typeof ngDevMode !== 'undefined' && !!ngDevMode;
```